docs/Training-Curriculum-Learning.md

```diff
@@ -2,68 +2,97 @@
-over time for an agent trained using PPO with the same set of training hyperparameters. 
+
+To see this in action, observe the two learning curves below. Each displays the reward
+over time for an agent trained using PPO with the same set of training hyperparameters.
 The difference is that one agent was trained using the full-height wall
 version of the task, and the other agent was trained using the curriculum version of
-the task. As you can see, without using curriculum learning the agent has a lot of 
-difficulty. We think that by using well-crafted curricula, agents trained using 
-reinforcement learning will be able to accomplish tasks otherwise much more difficult. 
- 
+the task. As you can see, without using curriculum learning the agent has a lot of
+difficulty. We think that by using well-crafted curricula, agents trained using
+reinforcement learning will be able to accomplish tasks otherwise much more difficult.
+
 ![Log](images/curriculum_progress.png)
 
 ## How-To
- 
-So how does it work? In order to define a curriculum, the first step is to decide which 
-parameters of the environment will vary. In the case of the Wall Area environment, what 
-varies is the height of the wall. We can define this as a reset parameter in the Academy 
-object of our scene, and by doing so it becomes adjustable via the Python API. Rather 
-than adjusting it by hand, we then create a simple JSON file which describes the 
-structure of the curriculum. Within it we can set at what points in the training process 
-our wall height will change, either based on the percentage of training steps which have 
-taken place, or what the average reward the agent has received in the recent past is. 
-Once these are in place, we simply launch learn.py using the `–curriculum-file` flag to 
-point to the JSON file, and PPO we will train using Curriculum Learning. Of course we can 
-then keep track of the current lesson and progress via TensorBoard.
 
+Each Brain in an environment can have a corresponding curriculum. These
+curriculums are held in what we call a metacurriculum. A metacurriculum allows
+different brains to follow different curriculums within the same environment.
+
+### Specifying a Metacurriculum
+
+We first create a folder inside `python/curricula/` for the environment we want
+to use curriculum learning with. For example, if we were creating a metacurriculum
+for Wall Jump, we would create the folder `python/curricula/wall-jump/`. We will place
+our curriculums inside this folder.
+
+### Specifying a Curriculum
+
+In order to define a curriculum, the first step is to decide which
+parameters of the environment will vary. In the case of the Wall Jump environment, what
+varies is the height of the wall. We define this as a `Reset Parameter` in the Academy
+object of our scene, and by doing so it becomes adjustable via the Python API. Rather
+than adjusting it by hand, we will create a simple JSON file which describes the
```

Comment:
remove simple
It is not obvious that the json files must have the names of the brains in the scene. This should be important.